features/lwipstack/lwip_tools.cpp

```diff
@@ -273,11 +273,8 @@ bool convert_mbed_addr_to_lwip(ip_addr_t *out, const nsapi_addr_t *in)
 
 #if LWIP_IPV4 && LWIP_IPV6
     if (in->version == NSAPI_UNSPEC) {
-#if IP_VERSION_PREF == PREF_IPV4
-        ip_addr_set_zero_ip4(out);
-#else
         ip_addr_set_zero_ip6(out);
```

Comment:
In case out is `ip4_addr` this probably crashes (overflow?). 
ip6_addr has u32_t[4] while ip4_addr has just u32_t.
So this check cannot be removed

Reply:
When LWIP_IPV4 and LWIP_IPV6 is enabled, both ip_addr_set_zero_ip4() and ip_addr_set_zero_ip6() api's will clear u32_t[4] bytes internally. Only difference is setting of IPADDR_TYPE. 
Additional reference in ip_addr.h, line 168 and 171 ip6_addr_set_zero() is used in both the cases to clear the bytes.
IPv4 u32_t and IPv6 u32_t[4] is declared in the union, size of it is always u32_t[4], hence overflow cannot occur.
So changed to call only ip_addr_set_zero_ip6() api in both the cases and IPADD_TYPE is set to IPADDR_TYPE_ANY

Reply:
> In case out is ip4_addr this probably crashes (overflow?).

`ip_addr_t` must be the v4/v6-union-with-type if we're in the `#if LWIP_IPV4 && LWIP_IPV6` block. `ip_addr_set_zero_ip6` is not defined unless we're in a config where it can take an IPv6 address.

> ip6_addr_set_zero() is used in both the cases to clear the byte

Indeed.

It's weird they don't have `ip_addr_set_zero_any()`. But to match their implementation of `ip_addr_set_zero_ip4/6`, I would change this line to `ip6_addr_set_zero` - the current version does "set the ip6-sized-address to 0, set the type to IPv6, set the type to any". Compiler will probably optimise out the double type setting, but may as well say what we mean.

Reply:
Will change it to ip6_addr_set_zero() to avoid double type setting.